### DISCUSSION TITLE
Maps in other (non-Mercator) projections

### DISCUSSION THREAD

**Discussion by MattBlissett** [onthegomap/planetiler#284] [Ideas]
Has anyone tried adapting this project to use maps in projections other than Web Mercator?

A few years ago I used OpenMapTiles' tools to generate the vector tiles on https://tile.gbif.org/ui/, in WGS84 plate carreé (EPSG:4326), Arctic LAEA (EPSG:3575) and Antarctic stereographic (EPSG:3031) projections, but planetiler seems much more efficient. 

These particular projections are more complicated to handle than some since EPSG:4326 has 2×1 tiles at zoom level 0 and its units are degrees rather than metres, and the polar projections don't wrap around.

**Comment by wipfli** [comment]
https://www.jasondavies.com/maps/transition/

**Comment by msbarry** [comment]
Not out of the box, but with some refactoring it might be possible...

Planetiler is specialized for generating tile pyramids where there is 1 z0 tile that splits into 4 z1 tiles, etc.  It would be possible to change how input coordinates map to those tiles. It's currently hard-coded in `getWorldX`/`getWorldY` `getWorldLat`/`getWorldLon` in [GeoUtils](https://github.com/onthegomap/planetiler/blob/main/planetiler-core/src/main/java/com/onthegomap/planetiler/geo/GeoUtils.java) but we could make it so you can swap that out with different logic, for example to do a polar map like https://tile.gbif.org/ui/3031/.

Changing so that there's more than 1 z0 tile would be more invasive though since that assumption is baked in more places.  Can clients like maplibre-gl handle 2 z0 tiles like that? A workaround would be to define a projection where the world maps to a rectangle that only fills the middle half of the z0 tile - from (0, 0.25) to (1, 0.75)

**Reply by MattBlissett** [reply]
Thanks!

Just to follow up in case others see this, although this isn't currently a priority for me: two Z0 tiles is used e.g. by https://cloud.maptiler.com/maps/basic-4326/, and before that one of the ArcGIS online maps, although that seems to have been taken down.
